docs/source/data/fetching-rest.mdx

```diff
@@ -89,75 +126,88 @@ const resolvers = {
+  context: async ({ req }) => {
+    const { cache } = server;
+    return {
+      token,
+      //highlight-start
+      dataSources: {
+        moviesAPI: new MoviesAPI({ cache, token }),
+      },
+      //highlight-end
+    };
+  },
+});
 ```
 
-You then extend the `RESTDataSource` class and implement whatever data-fetching methods your resolvers need. These methods can use built-in convenience methods (like `get` and `post`) to perform HTTP requests, helping you add query parameters, parse JSON results, and handle errors.
+</MultiCodeBlock>
 
-### Example
+If you want Apollo Server to use another cache backend, you can provide its constructor a _different_ cache object that implements the [`KeyValueCache` interface](https://github.com/apollographql/apollo-utils/tree/main/packages/keyValueCache#keyvaluecache-interface):
 
-Here's an example `RESTDataSource` subclass that defines two data-fetching methods, `getMovie` and `getMostViewedMovies`:
+```ts title="server.ts"
+const server = new ApolloServer({
+  typeDefs,
+  resolvers,
+  cache: new MyCustomKeyValueCache()
+});
+```
 
-```ts title="movies-api.js"
-const { RESTDataSource } = require('apollo-datasource-rest');
+### Using an external cache backend
 
-class MoviesAPI extends RESTDataSource {
-  constructor() {
-    // Always call super()
-    super();
-    // Sets the base URL for the REST API
-    this.baseURL = 'https://movies-api.example.com/';
-  }
+When running multiple instances of your server, you should use a shared cache backend. This enables one server instance to use the cached result from _another_ instance.
 
-  async getMovie(id) {
-    // Send a GET request to the specified endpoint
-    return this.get(`movies/${encodeURIComponent(id)}`);
-  }
+Apollo Server supports using [Memcached](https://memcached.org/), [Redis](https://redis.io/), or other cache backends via the [`keyv`](https://www.npmjs.com/package/keyv) package. For examples, see [Configuring external caching](../performance/cache-backends#configuring-external-caching).
 
-  async getMostViewedMovies(limit = 10) {
-    const data = await this.get('movies', {
-      // Query parameters
-      per_page: limit,
-      order_by: 'most_viewed',
-    });
-    return data.results;
-  }
-}
-```
+You can also choose to implement your own cache backend. For more information, see [Implementing your own cache backend](../performance/cache-backends#implementing-your-own-cache-backend).
 
-### HTTP Methods
+## HTTP Methods
 
-`RESTDataSource` includes convenience methods for common REST API request methods: `get`, `post`, `put`, `patch`, and `delete` ([see the source](https://github.com/apollographql/apollo-server/blob/main/packages/apollo-datasource-rest/src/RESTDataSource.ts#L154-L202)).
+`RESTDataSource` includes convenience methods for common REST API request methods: `get`, `post`, `put`, `patch`, and `delete` ([see the source](https://github.com/apollographql/datasource-rest/blob/main/src/RESTDataSource.ts#L163)).
```

Comment:
If we're linking to a specific line number, we _shouldn't_ reference `.../blob/main/...` since that's a moving target. If the code changes on main, your link becomes invalid.

To remedy this, when you select the line in GH you'll see 3 dots - you want copy permalink. You can also just replace `main` in your URL with the most recent commit hash.

https://github.com/apollographql/datasource-rest/blob/25862e18d8b35e324c150654c5686ed317b3fca8/src/RESTDataSource.ts#L163

Reply:
That's a great point, I always do wonder what the best way to do GH code links is. I'd rather not link folks to old commits, just in case you update things, but I can't think of a better way right now—so permalink it is!